addons/web/static/src/js/views/pivot/pivot_model.js

```diff
@@ -715,6 +712,23 @@ var PivotModel = AbstractModel.extend({
             this._traverseTree(root.children[i], f, arg1, arg2, arg3);
         }
     },
+    /**
+     * Updates the groupBy that the main[row | col] holds
+     * Because of expandHeader: l128, the list must be shared
+     * between the main element and its root
+     *
+     * @param {Object} The main abstract header before reload
+     * @param {Object} The main abstract header after reload
+     *
+     * @private
+     *
+     */
+    _updateGlobalGroupBys: function (old, main) {
+        var new_groupby_length = this._getHeaderDepth(main.root) - 1;
+        var new_groupby_list = old.root.groupbys.slice(0, new_groupby_length);
+        main.groupbys = new_groupby_list;
+        main.root.groupbys = new_groupby_list;
```

Comment:
i feel that we duplicate the information.  This is dangerous, because we may forget to synchronize them properly at some point.  Would it be possible to instead change the rest of the pivot model to only use the groupbys list from the main row/main col, and never the same list from the roots?  That would mean slighty changing the expandHeader method, for example

Reply:
I agree on the need to refactor, but I feel it needs to be well done, hence not as a bugfix.